### PR TITLE
RH-SSO Deployment doesn't reflect serviceAccountName

### LINKED ISSUE #569: RH-SSO Deployment doesn't reflect `serviceAccountName`
### Describe the bug

Changes introduced in https://github.com/keycloak/keycloak-operator/pull/357 are not reflected in RH-SSO.

### Version

15.1.0-18.0.2

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by vmuzikar**
Closes #569

### CODE REVIEW

**Comment by vmuzikar** [comment]
Also sneaked in a few fixes for `serviceAccoutName` as well.
1. `serviceAccoutName` is now reconciled, not just used at STS creation.
2. We don't assume the default `serviceAccoutName` at all. We just pass the empty string if not set in CR which is the default for the K8s client.

File: pkg/model/keycloak_deployment_test.go [hunk 1]
```diff
@@ -489,20 +489,22 @@ func testServiceAccountSet(t *testing.T, deploymentFunction createDeploymentStat
...
-
-	//If serviceAccountName is not set in the cr, then the serviceAccountName should be default
+	statefulSet := deploymentFunction(cr, dbSecret, nil)
 
 	//when
-	serviceAccountName := deploymentFunction(cr, dbSecret, nil).Spec.Template.Spec.ServiceAccountName
 
-	assert.Equal(t, "default", serviceAccountName)
+	//If serviceAccountName is set in the cr, is should manifest itself in the statefulset
+	cr.Spec.KeycloakDeploymentSpec.Experimental.ServiceAccountName = "test2"
```
Comment (DGuhr) [inline pkg/model/keycloak_deployment_test.go:501]: do we have e2e tests like for the new operator for this one? Would be interesting to know what happens when no ServiceAccountName is set (==nil or empty string i guess). I have too little context to evaluate the side effects, tbh, thats why I ask.

Comment (vmuzikar) [reply pkg/model/keycloak_deployment_test.go:501]: In any case, the default (no `serviceAccountName` set) is actually tested by all E2E tests as we don't set it anywhere. :)

Comment (DGuhr) [reply pkg/model/keycloak_deployment_test.go:501]: Thanks for the explanation. In that case I approve :)

**Review by pedroigor — APPROVED** [review]

**Pull request merged**